### PR TITLE
Better failure modes

### LINKED ISSUE #1305: Provide better failure modes when a connection error occurs
__Problem:__
Currently if VS is started up disconnected from the internet, the connection is simply removed from Team Explorer. 

__Solution ideas:__
We need to display a message with a "Retry" button or similar if a connection error happens, and similarly display a "You need to log in again" button or similar if the users credentials have changed.

### PR DESCRIPTION
**Pull request by grokys**
When a login fails at VS startup, currently we just remove the connection from the TE connect page and pretend it never existed. With this PR, we show a message and allow the user to retry (in the case of non-auth-related) errors or sign out:

General connection error:

image

Auth error:

image

In addition, an error is shown in the GitHub pane with a link to open the Team Explorer Connect section to fix the problem:

image

The UI needs some love from @donokuda , but this is ready to play with!

Fixes #1305

### CODE REVIEW

**Comment by meaghanlewis** [comment]
Tested this by changing my GitHub for Visual Studio credentials in the credential manager to be incorrect and by using the proxy server Burp.

@donokuda if you need any help testing your changes locally, let me know.

This LGTM, thanks @grokys! ✨

File: src/GitHub.TeamFoundation.14/Connect/GitHubConnectSection.cs [hunk 1]
```diff
@@ -249,6 +291,23 @@ void OnPropertyChange(object sender, PropertyChangedEventArgs e)
                 settings.IsExpanded = IsExpanded;
         }
 
+        void InvitationSectionPropertyChanged(object sender, PropertyChangedEventArgs e)
+        {
+            if (e.PropertyName == "IsVisible")
```
Comment (StanleyGoldman) [inline src/GitHub.TeamFoundation.14/Connect/GitHubConnectSection.cs]: `nameof` maybe?

File: src/GitHub.App/ViewModels/GitHubPane/GitHubPaneViewModel.cs [hunk 2]
```diff
@@ -410,26 +417,56 @@ async Task UpdateContent(ILocalRepositoryModel repository)
...
                 }
                 else
                 {
                     log.Debug("Found a a GitHub repository but not logged in: {CloneUrl}", repository?.CloneUrl);
                     Content = loggedOut;
                 }
             }
-            else
+            
+            if (notGitHubRepo)
```
Comment (StanleyGoldman) [inline src/GitHub.App/ViewModels/GitHubPane/GitHubPaneViewModel.cs]: `else if`?

Comment (grokys) [reply src/GitHub.App/ViewModels/GitHubPane/GitHubPaneViewModel.cs]: No, this flag is set in the preceding block.

File: src/GitHub.TeamFoundation.14/Connect/GitHubConnectSection.cs (hunk 1 above)
Comment (grokys) [reply src/GitHub.TeamFoundation.14/Connect/GitHubConnectSection.cs]: Yeah I was trying to stick with the style of the previous method which didn't use `nameof`, but you're right. I've converted this method and the previous to use `nameof`.

**Pull request merged**